core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -301,7 +305,16 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
       // - final_workflow_outputs_dir = submissions/final-outputs
       // - final_workflow_outputs_mode = "copy".
 
+      useCromwellGcpBatchBackend: Boolean = currentSettings
+        .collectFirst { case setting: UseCromwellGcpBatchBackendSetting =>
+          setting
+        }
+        .exists(_.config.enabled)
```

Comment:
What do you mean by a single invocation?

Reply:
I mean going from a `collectFirst` followed by a `exists`, to just an `exists`. I find it easier to read because I don't have to think about the type coming out of the `collectFirst`. I also like making it obvious that we are handling two different `false` cases - the settings exists and is set to false, or the setting doesn't exist.

Reply:
we should probably create a reusable utility method to retrieve a strongly-typed workspace setting from the settings list, a la:
```scala
  def findSetting[T](allSettings: List[WorkspaceSetting]): Option[T] = {
    allSettings.collectFirst {
      case found:T => found
    }
  }
```
if we had that, then this code could be a bit clearer, like:
```scala
useBatchSetting: Option[UseCromwellGcpBatchBackendSetting] = somwhereTheUtilityLives
        .findSetting[UseCromwellGcpBatchBackendSetting](currentSettings)
```
or
```scala
useCromwellGcpBatchBackend: Boolean = somwhereTheUtilityLives
        .findSetting[UseCromwellGcpBatchBackendSetting](currentSettings).exists(_.config.enabled)
```
you could use the same utility to calculate `separateSubmissionSetting` up at line 293, too.

I'd consider this a stretch goal/out of scope/does not block this PR. I don't want to scope-creep!

Reply:
Thanks! @davidangb I agree that refactoring would be great for clarity and reusability in the future, but I have implemented Janet's suggestion for now.